stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/EventsParser.kt

```diff
@@ -69,21 +71,37 @@ internal class EventsParser(
             if (!connectionEventReceived) {
                 if (event is ConnectedEvent) {
                     connectionEventReceived = true
-                    service.onConnectionResolved(event)
+                    onConnectionResolved(event)
                 } else {
-                    service.onSocketError(ChatNetworkError.create(ChatErrorCode.CANT_PARSE_CONNECTION_EVENT))
+                    onSocketError(ChatNetworkError.create(ChatErrorCode.CANT_PARSE_CONNECTION_EVENT))
                 }
             } else {
-                service.onEvent(event)
+                onEvent(event)
             }
         } else {
-            service.onSocketError(
-                ChatNetworkError.create(ChatErrorCode.CANT_PARSE_EVENT, eventResult.error().cause)
-            )
+            onSocketError(ChatNetworkError.create(ChatErrorCode.CANT_PARSE_EVENT, eventResult.error().cause))
         }
     }
 
     private fun handleErrorEvent(error: ErrorResponse) {
-        service.onSocketError(ChatNetworkError.create(error.code, error.message, error.statusCode))
+        onSocketError(ChatNetworkError.create(error.code, error.message, error.statusCode))
+    }
+
+    private fun onSocketError(error: ChatError) {
+        if (!closedByClient) {
```

Comment:
Can't we bring these checks earlier, to the `handleEvent` or even the `onMessage` functions? This way we're doing a bunch of parsing before we discard the result. If that's too early in some cases, maybe we can have checks in both places?